types/stylelint/index.d.ts

```diff
@@ -279,147 +279,159 @@ declare namespace stylelint {
-		'block-no-empty': CoreRule<true, { ignore: 'comments'[] }>;
-		'color-function-notation': CoreRule<'modern' | 'legacy', { ignore: 'with-var-inside'[] }>;
+		'at-rule-property-required-list': CoreRule<Record<string, OneOrMany<string>>>;
+		'block-no-empty': CoreRule<true, { ignore: OneOrMany<'comments'> }>;
+		'color-function-notation': CoreRule<
+			'modern' | 'legacy',
+			{ ignore: OneOrMany<'with-var-inside'> }
+		>;
 		'color-hex-alpha': CoreRule<'always' | 'never'>;
 		'color-hex-length': CoreRule<'short' | 'long'>;
 		'color-named': CoreRule<
 			'never' | 'always-where-possible',
-			{ ignoreProperties: StringsOrRegexes; ignore: 'inside-function'[] }
+			{ ignoreProperties: OneOrMany<StringOrRegex>; ignore: OneOrMany<'inside-function'> }
 		>;
 		'color-no-hex': CoreRule<true>;
 		'color-no-invalid-hex': CoreRule<true>;
 		'comment-empty-line-before': CoreRule<
 			'always' | 'never',
 			{
-				except: 'first-nested'[];
-				ignore: ('stylelint-commands' | 'after-comment')[];
-				ignoreComments: StringsOrRegexes;
+				except: OneOrMany<'first-nested'>;
+				ignore: OneOrMany<'stylelint-commands' | 'after-comment'>;
+				ignoreComments: OneOrMany<StringOrRegex>;
 			}
 		>;
 		'comment-no-empty': CoreRule<true>;
 		'comment-pattern': CoreRule<StringOrRegex>;
 		'comment-whitespace-inside': CoreRule<'always' | 'never'>;
-		'comment-word-disallowed-list': CoreRule<StringOrRegex | StringsOrRegexes>;
+		'comment-word-disallowed-list': CoreRule<OneOrMany<StringOrRegex>>;
 		'custom-media-pattern': CoreRule<StringOrRegex>;
 		'custom-property-empty-line-before': CoreRule<
 			'always' | 'never',
 			{
-				except: ('first-nested' | 'after-comment' | 'after-custom-property')[];
-				ignore: ('after-comment' | 'first-nested' | 'inside-single-line-block')[];
+				except: OneOrMany<'first-nested' | 'after-comment' | 'after-custom-property'>;
+				ignore: OneOrMany<'after-comment' | 'first-nested' | 'inside-single-line-block'>;
 			}
 		>;
 		'custom-property-no-missing-var-function': CoreRule<true>;
 		'custom-property-pattern': CoreRule<StringOrRegex>;
 		'declaration-block-no-duplicate-custom-properties': CoreRule<
 			true,
-			{ ignoreProperties: StringsOrRegexes }
+			{ ignoreProperties: OneOrMany<StringOrRegex> }
 		>;
 		'declaration-block-no-duplicate-properties': CoreRule<
 			true,
 			{
-				ignore: Array<
+				ignore: OneOrMany<
 					| 'consecutive-duplicates'
 					| 'consecutive-duplicates-with-different-values'
 					| 'consecutive-duplicates-with-different-syntaxes'
 					| 'consecutive-duplicates-with-same-prefixless-values'
 				>;
-				ignoreProperties: StringsOrRegexes;
+				ignoreProperties: OneOrMany<StringOrRegex>;
 			}
 		>;
 		'declaration-block-no-redundant-longhand-properties': CoreRule<
 			true,
-			{ ignoreShorthands: StringsOrRegexes; ignoreLonghands: string[] }
+			{
+				ignoreShorthands: OneOrMany<StringOrRegex>;
+				ignoreLonghands: string[];
```

Comment:
[note] Out-of-scope. `ignoreLonghands` should be `OneOrMany<StringOrRegex>`. 😅

Reply:
https://github.com/stylelint/stylelint/pull/7611#discussion_r1568916551